lnrpc/invoicesrpc/addinvoice.go

```diff
@@ -532,38 +557,98 @@ func chanCanBeHopHint(channel *channeldb.OpenChannel, cfg *AddInvoiceConfig) (
 
 // addHopHint creates a hop hint out of the passed channel and channel policy.
 // The new hop hint is appended to the passed slice.
-func addHopHint(hopHints *[]func(*zpay32.Invoice),
-	channel *channeldb.OpenChannel, chanPolicy *channeldb.ChannelEdgePolicy) {
+func addHopHint(hopHints *[][]zpay32.HopHint,
```

Comment:
nit: instead of passing a pointer to slice it may be worth to follow the same pattern as `append()`

```
addHopHint(hopHints [][]zpay32.HopHint,channel *HopHintInfo, chanPolicy *channeldb.ChannelEdgePolicy) [][]zpay32.HopHint {
    // modify hopHints
    
    return hopHints
}
```

Reply:
I also don't really like the pointer here, but thought "if it ain't broke, don't fix it" in the hopes of minimizing the refactor. Happy to change if its grating you.

Reply:
Also there's no need for the pointer as the slice is passed by reference.

Reply:
@bhandras  I think that would work if it was not an array of arrays because we create the array with enough capacity + append/modify up only to that capacity ([Go playground test](https://go.dev/play/p/qgA8VweZK9t)). 

Even so, better make it straight forward for people like me :sweat_smile: 

Also, why are we using an array of arrays here when `addHopHint` adds arrays of only one item? 

```
 	*hopHints = append(*hopHints, []zpay32.HopHint{hopHint})
```

Reply:
> Also, why are we using an array of arrays here when addHopHint adds arrays of only one item?

I believe that we could theoreticaly have a route specified by a set of hop hints (eg `A --- B ---C` and we provide one for each, but lnd doesn't support chained hints right now. All calling functions will need `[][]` because they have to apply to `zpay32` invoices so I'd be in favor of leaving as is? Otherwise every caller needs to loop through and create `[][]hints`?